app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala

```diff
@@ -67,17 +67,24 @@ class SubscriptionController @Inject()(
       case hasNonEmptyEnrolments(_) =>
         Future.successful(Redirect(routes.BusinessIdentificationController.showCreateNewAccount()))
       case _ =>
-        mark("Count-Subscription-CleanCreds-Success")
-        withInitialDetails { details =>
-          Future.successful {
+        val details = for {
+          initialDetails <- sessionStoreService.fetchInitialDetails
+          amlsDetails    <- sessionStoreService.fetchAMLSDetails
+        } yield (initialDetails, amlsDetails)
+
+        details.map {
+          case (Some(initialDetails), mayBeAmlsDetails) =>
```

Comment:
if the amlsDetails are a "maybeAmlsDetails" then you should yield optAmlsDetails